src/datasource/zabbix/connectors/zabbix_api/zabbixAPIConnector.ts

```diff
@@ -793,7 +793,7 @@ export class ZabbixAPIConnector {
         triggers = triggers.length;
       }
       // When version is 7.0.0 or higher, groups are returned as hostgroups
-      if (semver.gte(this.version, '7.0.0')) {
+      if (semver.gte(this.version, '7.0.0') && !count) {
         triggers.forEach((trigger) => {
           trigger.groups = trigger.hostgroups;
         });
```

Comment:
i noticed this block of code is repeated in a few places. would it make sense to extract it into a separate helper function? so that we can ensure consistency and reduce the risk of these falling out of sync in the future.

just a thought, feel free to ignore.